### ISSUE TITLE
Incorrect calculation of McCabe for when expression?

### ISSUE THREAD

**Issue by arturbosch** [arturbosch/detekt#396]
Detekt does not care how many branches a when expression has. The complexity is incremented for the existence of a when expression + and the existence of the else branch.

For following example detekt calculates the McCabe complexity of 2 and not 5 (number of branches). 5 seems reasonable due to the number of branches, but the subjective complexity is not much higher than of an simple if statement (due to clean structure of when expressions? readability)

```kotlin
when(number) {
    1 -> println("One")
    2 -> println("Two")
    3 -> println("Three")
    4 -> println("Four")
    else -> println("Big Number")
}
```

Should detekt calculate 5 in such cases? Should detekt check how many statements each branch has and only increment the complexity for each block?

**Comment by schalkms** [comment]
In my opinion it should report 5 in this case.
